src/wp-admin/includes/class-wp-site-health.php

```diff
@@ -1498,6 +1498,10 @@ public function get_test_is_in_debug_mode() {
 	 * @return array The test results.
 	 */
 	public function get_test_https_status() {
+		// Enforce fresh HTTPS detection results. This is normally invoked by using cron, but for Site Health it should
+		// always rely on the latest results.
+		wp_update_https_detection_errors();
```

Comment:
Great!

Reply:
Should this be an async test now so we don't slow down page load?